packages/babel-core/src/config/option-manager.js

```diff
@@ -327,10 +469,16 @@ function createDescriptor(
     ownPass?: boolean,
   },
 ): BasicDescriptor {
+  let name;
   let options;
   let value = pair;
   if (Array.isArray(value)) {
-    [value, options] = value;
+    if (value.length === 3) {
+      // $FlowIgnore - Flow doesn't like the multiple tuple types.
```

Comment:
is the length check necessary here? you could just unpack the tuple here as is and the `name` would simply stay `undefined`, although we shouldnt access out of bounds elements, so maybe it would be reasonable to have only 1 tuple type for this, where the last element would be a maybe type AND it would require explicit third element of null/undefined

Reply:
> where the last element would be a maybe type AND it would require explicit third element of null/undefined

Reply:
Hm, maybe you could just normalize the user's input as soon as possible to use consistent type in the core and avoid `$FlowIgnore`?

Reply:
I can't do that _currently_ because copying the input object would invalidate some of the plugin caching, that said I do think it is confusing right now. Perhaps something to revisit in the future?

Reply:
In that case it is certainly not worth changing right now.